styles/scss/mixins/_underline-text.scss

```diff
@@ -0,0 +1,109 @@
+/// @param {timing-function} $timing-function [cubic-bezier(.79, .01, .22, .99)] - Execution speed function.
+/// @param {hex} $color [currentColor] - Color of the line.
+/// @param {boolean} $wrapper [false] - If wrapper is used as hover reference.
+/// @param {string} $element [.underline-text] - Element on which you want underline.
+/// @param {string} $state [hover] - The state in which it becomes underline element.
+/// @example
+/// .test {
+///   @include underline-text();
+/// }
+/// .test2 {
+///   @include underline-text(10, .7s, ease-in, #132031, true, a.link, hover);
+/// }
+/// @output
+/// ```scss
+/// .test {
+///   display: inline;
+///   transition: background-size 0.5s cubic-bezier(0.79, 0.01, 0.22, 0.99) 0s, background-position 0s step-end 0.5s;
+///   text-decoration: none;
+///   background-image: linear-gradient(transparent 90%, currentColor 90%, currentColor 100%);
+///   background-repeat: no-repeat;
+///   background-position-y: bottom;
+///   background-size: 0% 100%;
+///   &:hover {
+///     background-position-x: right;
+///     background-position-y: bottom;
+///     background-size: 100% 100%;
+///   }
+/// }
+/// .test2 {
+///   & .link {
+///     display: inline;
+///     transition: background-size 0.7s ease-in 0s, background-position 0s step-end 0.7s;
+///     text-decoration: none;
+///     background-image: linear-gradient(transparent 95%, #132031 95%, #132031 100%);
+///     background-repeat: no-repeat;
+///     background-position-y: bottom;
+///     background-size: 0% 100%;
+///   }
+///   &:hover .link {
+///     background-position-x: right;
+///     background-position-y: bottom;
+///     background-size: 100% 100%;
+///   }
+/// }
+/// ```
+///
+
+
+@mixin underline-text($thickness: 10%, $duration: 0.5s, $timing-function: cubic-bezier(0.79, 0.01, 0.22, 0.99), $color: currentColor, $wrapper: false, $element: '.underline-text', $state: 'hover') {
+  $thickness: 100% - $thickness;
+
+  $underline-inactive-state: (
+    display: inline,
+
+    transition: (background-size $duration $timing-function 0s, background-position 0s step-end $duration),
+    text-decoration: none,
+
+    background-image: linear-gradient(transparent $thickness, $color $thickness, $color 100%),
+    background-repeat: no-repeat,
+    background-position-y: bottom,
+    background-size: 0% 100%,
+  );
+
+  $underline-active-state: (
+    background-position-x: right,
+    background-position-y: bottom,
+    background-size: 100% 100%,
+  );
+
+  @if $wrapper {
+    text-decoration: none;
+
+    #{$element} {
+      @include for-each-attribute($underline-inactive-state);
+    }
+
+    &:#{$state} {
+      #{$element} {
+        @include for-each-attribute($underline-active-state);
+      }
+    }
+  }
+
+  @else {
+    @include for-each-attribute($underline-inactive-state);
+
+    &:#{$state} {
+      @include for-each-attribute($underline-active-state);
+    }
+  }
+}
+@mixin for-each-attribute($map) {
```

Comment:
Hm I think this can be extracted in a separate standalone mixin @gabriel-glo any thoughts?

Reply:
Definitely a separate mixin. This can be reused on other places.
I'm thinking this could also be turned into a scss function.

Reply:
I would say that it can't be a function because of the @return paramater. I didn't find a way to return all of the parameters at once. 
@gabriel-glo do you maybe know how to?